bootloader/src/main.c

```diff
@@ -411,6 +411,12 @@ int main(void)
     }
     else if (USB_DFU_MODE == 0)
     {
+#if HAL_PLATFORM_RTL872X
+        if (!is_application_valid(ApplicationAddress, NULL)) {
```

Comment:
We should not call this all the time. Let's set DCT flag or something like that if we detect valid system-part at least once, and not call this anymore.

Reply:
What about?
```diff
--- a/bootloader/src/main.c
+++ b/bootloader/src/main.c
@@ -445,6 +445,11 @@ int main(void)
             uint32_t stack = *(volatile uint32_t*)entry;
             jump_to_system(addr, stack);
         }
+#if HAL_PLATFORM_RTL872X
+        else if (FLASH_AddMfgSystemModuleSlot() == 0) {
+            HAL_Core_System_Reset_Ex(RESET_REASON_UPDATE, 0, NULL);
+        }
+#endif
 #if !HAL_PLATFORM_NRF52840 && !HAL_PLATFORM_RTL872X
         else
         {
```
This way we don't have to complicate the DCT layout and just at the cost of reseting device one more time.